pkg/netbox/api/prefix_claim.go

```diff
@@ -80,6 +91,52 @@ func validatePrefixLengthOrError(prefixClaim *models.PrefixClaim, prefixFamily i
 	return nil
 }
 
+func (r *NetboxClient) GetAvailablePrefixByParentPrefixSelector(customFields map[string]string, tenant, prefixLength string) ([]*models.Prefix, error) {
+	var conditions func(co *runtime.ClientOperation)
+	customFieldEntries := make([]CustomFieldEntry, 0, len(customFields))
+	for k, v := range customFields {
+		customFieldEntries = append(customFieldEntries, CustomFieldEntry{
+			key:   k,
+			value: v,
+		})
+	}
+	conditions = newCustomFieldStringFilterOperation(customFieldEntries)
+
+	list, err := r.Ipam.IpamPrefixesList(ipam.NewIpamPrefixesListParams(), nil, conditions)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: find a better way?
+	if list.Payload.Count != nil && *list.Payload.Count == 0 {
+		return nil, nil
+	}
+
+	prefixes := make([]*models.Prefix, 0)
+	for _, prefix := range list.Payload.Results {
+		if prefix.Prefix != nil {
+			// if we can allocate a prefix from it, we can take it as a parent prefix
+			if _, err := r.getAvailablePrefixByPrefix(tenant, *prefix.Prefix, prefixLength); err == nil {
+				prefixes = append(prefixes, &models.Prefix{
+					Prefix: *prefix.Prefix,
+				})
+			}
+		}
+	}
+
+	return prefixes, nil
+}
+
+func (r *NetboxClient) getAvailablePrefixByPrefix(tenant, prefix, prefixLength string) (*models.Prefix, error) {
```

Comment:
Maybe it's all about context, but it is not clear to me what this method does based on its name

Reply:
@CapiSSS  I renamed it to `checkPrefixAvailbility`. Does this help? :)

Reply:
Thanks for checking @CapiSSS!